# Manuscript code repository
This repository contains the packages used, commands issued, and files to generate figures for the manuscript titled "Quorum-sensing synthase mutations re-calibrate autoinducer concentrations in clinical isolates of Pseudomonas aeruginosa to enhance pathogenesis". The goal of this repository is to increase transparency and reproducibility.

# Genome assembly and annotation
Data generating commands used for reference genome assembly and annotation:
- `genome_assembly_workflow.md` markdown file containing the commands issued to assemble the genomes used in this study
- `genome_annotation_workflow.md` markdown file containing the commands issued to annotate the genomes used in this study

# Phylogenetic analysis
Input files and commands issued to generate phylogenetic trees:
- `RhlI_acute_isolate_workflow.md` markdown file containing the commands issued to generate figure 1A
- `LasI_orthologue_workflow.md` markdown file containing the commands issued to generate supplemental figure 1B
- `RhlI_PA_workflow.md` markdown file containing the commands issued to generate figure X

# Citation information
List of packages, versions used, citations, and relevant links.
 
```
TrimGalore v0.6.7
https://github.com/FelixKrueger/TrimGalore
https://www.bioinformatics.babraham.ac.uk/index.html
doi.org/10.5281/zenodo.7598955

SAMtools v1.1.2
https://github.com/samtools/
https://www.htslib.org/
Li, H. et al. The Sequence Alignment/Map format and samtools. Bioinformatics 25, 2078 (2009).
doi:10.1093/bioinformatics/btp352

BCFtools v1.1.0.2
https://github.com/samtools/bcftools
Danececk, P. et al. Twelve years of SAMtools and BCFtools. Gigascience 10, 1-4 (2021).
doi:10.1093/gigascience/giab008

Cutadapt vx.x.x
https://github.com/marcelm/cutadapt
Martin, M. Cutadapt removes adapter sequences from high-throughput sequencing reads. EMBnet. journal 17, 10-12 (2011).
doi:10.14806/ej.17.1.200

Prokka v1.14
https://github.com/tseemann/prokka
Seemann, T. Prokka: rapid prokaryotic genome annotation. Bioinformatics 30, 2068-2069 (2014).
doi:10.1093/bioinformatics/btu153

MUSCLE v.3.8.1551
http://www.drive5.com/muscle
Edgar, R.C. MUSCLE: multiple sequence alignment with high accuracy and high throughput. Nucleic acids research 32, 1792–1797 (2004).
doi:10.1186/1471-2105-5-113

trimAL v.1.4.1
https://github.com/inab/trimal
Capella-Gutiérrez, S., Silla-Martínez, J. M. & Gabaldón, T. trimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses. Bioinformatics 25, 1972–1973 (2009).
doi:10.1093/bioinformatics/btp348

IQ-TREE v1.6.12
http://www.iqtree.org/
Nguyen, L. T., Schmidt, H. A., Von Haeseler, A. & Minh, B. Q. IQ-TREE: A Fast and Effective Stochastic Algorithm for Estimating Maximum-Likelihood Phylogenies. Mol Biol Evol 32, 268–274 (2015).
doi:10.1093/molbev/msu300

iTOL v.6.7.1
https://itol.embl.de/
Letunic, I. & Bork, P. Interactive Tree Of Life (iTOL) v5: an online tool for phylogenetic tree display and annotation. Nucleic Acids Res 49, W293–W296 (2021).
doi:10.1093/nar/gkab301
```
